File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsLearnerProfile.swift
import Foundation

// MARK: - Learner profile (LP10)

typealias LearnerProfileFacetKey = String

struct LearnerProfile: Codable, Equatable {
    var status: String
    var lastComputedAt: String?
    var facets: [LearnerProfileFacetSummary]
}

struct LearnerProfileFacetSummary: Codable, Equatable, Identifiable {
    var facetKey: LearnerProfileFacetKey
    var state: String
    var summary: [String: JSONValue]
    var confidence: Double
    var computedVersion: Int
    var updatedAt: String

    var id: String { facetKey }
}

struct LearnerProfileInsight: Codable, Equatable, Identifiable {
    var insightKey: String
    var label: String
    var value: [String: JSONValue]
    var confidence: Double
    var salience: Double
    var evidence: [LearnerProfileEvidenceRow]?

    var id: String { insightKey }
}

struct LearnerProfileEvidenceRow: Codable, Equatable, Identifiable {
    var sourceKind: String
    var sourceTable: String
    var observationCount: Int
    var courseId: String?
    var windowStart: String?
    var windowEnd: String?
    var contribution: Double?

    var id: String { "\(sourceKind)-\(sourceTable)-\(observationCount)-\(courseId ?? "")" }
}

struct LearnerProfileResponse: Codable {
    var profile: LearnerProfile?
}

struct LearnerProfileFacetDetailResponse: Codable {
    var facet: LearnerProfileFacetSummary?
    var insights: [LearnerProfileInsight]?
}

struct LearnerProfileControlResponse: Codable {
    var status: String?
}

typealias LearnerProfileEvidenceMap = [String: [LearnerProfileEvidenceRow]]

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsLearnerProfile.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

/** Learner profile models (LP10). */
@Serializable
data class LearnerProfile(
    val status: String = "insufficient_data",
    val lastComputedAt: String? = null,
    val facets: List<LearnerProfileFacetSummary> = emptyList(),
)

@Serializable
data class LearnerProfileFacetSummary(
    val facetKey: String,
    val state: String,
    val summary: JsonObject = JsonObject(emptyMap()),
    val confidence: Double = 0.0,
    val computedVersion: Int = 0,
    val updatedAt: String = "",
)

@Serializable
data class LearnerProfileInsight(
    val insightKey: String,
    val label: String,
    val value: JsonObject = JsonObject(emptyMap()),
    val confidence: Double = 0.0,
    val salience: Double = 0.0,
    val evidence: List<LearnerProfileEvidenceRow>? = null,
)

@Serializable
data class LearnerProfileEvidenceRow(
    val sourceKind: String,
    val sourceTable: String,
    val observationCount: Int,
    val courseId: String? = null,
    val windowStart: String? = null,
    val windowEnd: String? = null,
    val contribution: Double? = null,
)

@Serializable
data class LearnerProfileResponse(
    val profile: LearnerProfile? = null,
)

@Serializable
data class LearnerProfileFacetDetailResponse(
    val facet: LearnerProfileFacetSummary? = null,
    val insights: List<LearnerProfileInsight>? = null,
)

@Serializable
data class LearnerProfileControlResponse(
    val status: String? = null,
)